### PR TITLE
Added checking that exceptions were thrown in the python tests.

### PR DESCRIPTION
**Pull request by VolodymyrBabaiev**

### CODE REVIEW

File: qa/sc_ft_limit_fork.py [hunk 1]
```diff
@@ -91,11 +91,16 @@ def run_test(self):
...
         generate_next_blocks(sc_node, "first node", 1)
+
         # try to do another FT with 50 satoshi, assert it fails
+        exception_thrown = False
         try:
             self.do_ft(_50cent)
         except Exception as e:
+            exception_thrown = True
             assert_equal('16: bad-sc-tx-not-applicable', e.error['message'])
+        finally:
```
Comment (saratnt) [inline qa/sc_ft_limit_fork.py:102]: Same comment as in qa/sc_evm_storage_recovery.py file

File: qa/sc_storage_recovery_with_csw.py [hunk 2]
```diff
@@ -341,15 +341,17 @@ def run_test(self):
...
         try:
             # restart SC2
             self.forgeBlockAndCheckSync()
-
         except Exception as e:
+            exception_thrown = True
             logging.info("Expected exception caught during negative testing: " + str(e))
             logging.info("Stopping SC2")
             stop_sc_node(sc_node2, 1)
+        finally:
```
Comment (saratnt) [inline qa/sc_storage_recovery_with_csw.py:353]: Same comment as in qa/sc_evm_storage_recovery.py file

File: qa/sc_storage_recovery_without_csw.py [hunk 3]
```diff
@@ -318,15 +318,17 @@ def run_test(self):
...
         try:
             # restart SC2
             self.forgeBlockAndCheckSync()
-
         except Exception as e:
+            exception_thrown = True
             logging.info("Expected exception caught during negative testing: " + str(e))
             logging.info("Stopping SC2")
             stop_sc_node(sc_node2, 1)
+        finally:
```
Comment (saratnt) [inline qa/sc_storage_recovery_without_csw.py:330]: Same comment as in qa/sc_evm_storage_recovery.py file

File: qa/sc_evm_test_storage_contract.py [hunk 4]
```diff
@@ -94,7 +94,6 @@ def run_test(self):
         was_exception = False
         test_message = 'This is a message'
         try:
-            was_exception = False
             self.__set_storage_value(smart_contract, smart_contract_address, self.evm_address, test_message,
                                      static_call=True, generate_block=False)
         except EvmExecutionError as err:
```
Comment (saratnt) [inline qa/sc_evm_test_storage_contract.py:99]: Consider to remove `finally` clause and `was_exception` variable and to put a fail method in the `except` clause instead.

File: qa/sc_closed_forger.py [hunk 5]
```diff
@@ -194,6 +195,7 @@ def run_test(self):
         except SCAPIException as e:
             logging.info("Expected SCAPIException: " + e.error)
             error_occur = True
+
         assert_true(error_occur, "Try to send an openStake transaction with empty output proposition")
```
Comment (saratnt) [inline qa/sc_closed_forger.py:199]: Consider to remove `assert_true(error_occur...` and to put a `fail` after `createOpenStakeTransaction`. This way, calling assert_true can be avoided when SCAPIException is actually thrown.
The same comment can be applied in other files when the `except` clause doesn't catch a generic Exception, as in this case.

**Review by saratnt — APPROVED** [review]
LGTM, I have just some minor comments.

File: qa/sc_evm_test_storage_contract.py (hunk 4 above)
Comment (VolodymyrBabaiev) [reply qa/sc_evm_test_storage_contract.py:98]: Done.

**Review by i-Alex — APPROVED** [review]
Looks good,
rebase on top of `dev_evm` one more time and then we will merge

**Pull request merged**